Update containerd

ijc opened the pull request:
**- What I did**

Updated `containerd` from https://github.com/containerd/containerd/commit/c215531a8f63a98a69134e804fea4ee6d354bb90 to https://github.com/containerd/containerd/commit/8e1a04ff9ee302ae3b37013c42e8486465372cd9 and `runc` from https://github.com/opencontainers/runc/commit/ac50e77bbb440dcab354a328c79754e2502b79ca to https://github.com/opencontainers/runc/commit/429a5387123625040bacfbb60d96b1cbd02293ab

This is WIP because in theory it ought to be updated to include the result of merging https://github.com/moby/tool/pull/102, although I'm not seeing any breakage in practice.

**- How I did it**

Updated the alpine base package with the new container hash then rebase runc and containerd onto that and updated the runc hash.

**- How to verify it**

Build  an assembly (e.g. toplevel `linuxkit.yml`) and run it.

**- Description for the changelog**
Updated to `containerd` https://github.com/containerd/containerd/commit/8e1a04ff9ee302ae3b37013c42e8486465372cd9 and `runc` https://github.com/opencontainers/runc/commit/429a5387123625040bacfbb60d96b1cbd02293ab

**- A picture of a cute animal (not mandatory but encouraged)**
[![](https://upload.wikimedia.org/wikipedia/en/2/22/New_Age_of_Terror.jpg "Hirax, The New Age of Terror")](https://en.wikipedia.org/wiki/The_New_Age_of_Terror)

ijc commented:
@justincormack not sure if we actually want to wait for https://github.com/moby/tool/pull/102, I didn't see any breakage without it, but maybe CI will disagree. In any case unless there is something seriously wrong with https://github.com/moby/tool/pull/102 which requires major reworking there seems no harm in waiting here

justincormack commented:
The pointer changes don't affect the Json, platform being deleted should
not matter, and we were not setting oom due to it being moved, so I can't
see any reason why it should be an issue, which makes it easier...

On 6 Jul 2017 4:43 pm, "Ian Campbell" <notifications@github.com> wrote:

@justincormack <https://github.com/justincormack> not sure if we actually
want to wait for moby/tool#102 <https://github.com/moby/tool/pull/102>, I
didn't see any breakage without it, but maybe CI will disagree. In any case
unless there is something seriously wrong with moby/tool#102
<https://github.com/moby/tool/pull/102> which requires major reworking
there seems no harm in waiting here

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/linuxkit/linuxkit/pull/2159#issuecomment-313417219>,
or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AAdcPHNHY5Iltqz7T2jDvpnzyZMPH2Jvks5sLPKIgaJpZM4OPvEo>
.

ijc commented:
WRT not seeing any breakage I don't think any of the three changes outlined in https://github.com/moby/tool/pull/102 would cause a problem.

* `Platform` has been removed: even if it is present new `containerd` just ignores it (through the default behaviour of the Go JSON library).
* `Process` is now a pointer: this doesn't propagate into the JSON representation, so the two sides needn't agree
* `OOMScoreAdj` has moved into `Process`, from `Linux.Resources` (resolving a
TODO here): it was commented out before, nothing in this tree seems to use it

I still guess we may as well wait for https://github.com/moby/tool/pull/102 or we'll just need a second PR.

ijc commented:
Looks like we raced to the same conclusion!

justincormack commented:
Merged the tool PR as there is some internet under the Channel...

deitch commented:
Slightly off-topic, but I do see how many files you had to make the exact same update to `.yml` files. That composition of files (or other methods like combined `init` etc.) look ever more appealing.

ijc commented:
Pushed the tool update and dropped the WIP

I also threw in the helper script I've been using to do the bulk updates, we discussed it on slack weeks ago and decided it was useful enough even if very trivial to bung into `scripts` but I never got round to sneaking it in somewhere.

ijc commented:
Build error in CI, need to investigate.

That did remind me that I forgot to update linuxkit's own `vendor.conf`, I'll do that too

rn commented:
`moby` failed to build:
```
[+  3m 53s] # github.com/moby/tool/vendor/github.com/opencontainers/runc/libcontainer/user
[+  3m 53s] ../../vendor/github.com/opencontainers/runc/libcontainer/user/lookup.go:44: undefined: unix.Getuid
[+  3m 53s] ../../vendor/github.com/opencontainers/runc/libcontainer/user/lookup.go:92: undefined: unix.Getgid
[+  4m 12s] Makefile:31: recipe for target 'tmp_moby_bin.tar' failed
```

rn commented:
@deitch there is also https://github.com/moby/tool/issues/101

ijc commented:
Weirdly it builds for me.

ijc commented:
reproduced in a fresh $GOPATH, it's `make test-cross` which fails.

ijc commented:
*sigh* didn't have the right branch checkedout in my tree any more 😊

deitch commented:
> @deitch there is also moby/tool#101

Doesn't that make the build local-user-dependent (and thus non-reproducible)?

ijc commented:
Please can we discuss https://github.com/moby/tool/issues/101 on that ticket (or anywhere other than here, really).

ijc commented:
Updated to newer containerd as well as moby tool update to include portability fixes.

ijc commented:
Updated to pull in my fix for #2131 from in progress PR https://github.com/containerd/containerd/pull/1141 as discussed IRL.

Note that the new version of containerd changes `ctr exec --tty «container» «command»` into `ctr exec --tty --exec-id some-string «container» «command»`. If you forget the `--exec-id` then it will hang silently (see https://github.com/containerd/containerd/issues/1161).

This will conflict with #2191. Happy to rebase but it might take a moment...

riyazdf commented on projects/compose/compose-dynamic.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:b1766e4c4c09f63ac4925a6e4612852a93f7e73b
```
it's probably safe to bump containerd here, if we'd like to.  cc @deitch

riyazdf commented on projects/compose/compose-static.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:b1766e4c4c09f63ac4925a6e4612852a93f7e73b
```
same comment about updating containerd cc @deitch

riyazdf commented on projects/wireguard/wireguard.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:cbd7ae748f0a082516501a3e914fa0c924ee941e
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:f1130450206d4f64f0ddc13d15bb68435aa1ff61
```
same comment on containerd here, I can vouch for this

riyazdf commented on scripts/update-component-sha.sh:
```diff
@@ -0,0 +1,10 @@
+old=$1
+new=$2
+
+git grep -l "$old" | xargs sed -i -e "s,$old,$new,g"
```
did you want to include this in this PR?  It's fine with me

ijc replied:
Yes, bit out of place (other than the fact I did a lot of bulk updates in this PR) but figured it didn't deserve its own PR and I've been meaning to submit it for ages. https://github.com/linuxkit/linuxkit/pull/2159#issuecomment-313427313

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/011_tcp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
should we bump containerd here?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/010_tcp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
should we bump containerd here?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/020_tcp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
should we bump containerd here?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/021_tcp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
should we bump containerd here?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/030_tcp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/040_tcp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/050_udp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/051_udp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/060_udp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/061_udp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/070_udp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/000_4.4.x/080_udp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/010_tcp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/011_tcp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/020_tcp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/021_tcp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/030_tcp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/040_tcp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/050_udp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/051_udp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/060_udp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/061_udp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/070_udp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/001_4.9.x/080_udp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/010_tcp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/011_tcp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/080_udp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/070_udp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/061_udp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/060_udp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/051_udp4_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/050_udp4_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/040_tcp6_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/030_tcp4_loopback/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/021_tcp6_veth_reverse/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

riyazdf commented on test/cases/020_kernel/110_netns/003_4.11.x/020_tcp6_veth/test-netns.yml:
```diff
@@ -3,7 +3,7 @@ kernel:
   - linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:3bf218c20d9d5cf014f0223f37235e6d9dd8d904
   - linuxkit/containerd:c977f27c234d55b85172813b8451f67ea86be4a3
```
bump containerd?

justincormack commented:
We need to merge this as highest priority as it turned out the moby update did break linuxkit see https://github.com/moby/tool/issues/110 and also got other reports. (as it omits the OS, arch)

ijc commented:
Rebased and updated to pull in containerd v1.0.0-alpha0 which was released last night.

I didn't yet make the projects updates which @riyazdf suggested since I wasn't sure in each case if `init` would also require updates to match. I suspect the answer is yes.

What I'm considering doing instead is updating `runc` and `containerd` only for projects which are using `linuxkit/init:14a38303ee9dcb4541c00e2b87404befc1ba2083` (per the top-level `linuxkit.yml`) and reverting any other updates done under projects here. That would mean updating only:
* `projects/kubernetes`
* `projects/miragesdk`
* `projects/shiftfs`
* `projects/swarmd`
and not updating the others. Compared to the current state of this PR that means not touching:
* `projects/compose`
* `projects/etcd`
* `projects/image-namespace`
* `projects/logging`
* `projects/okernel`
* `projects/wireguard`.

NB -- for all I know they have already been broken by similar previous bulk updates.

ijc commented:
Hrm, there is also all the test cases :-/.

rn commented:
I think updating only the projects which are on the same hashes as `./linuxkit.yml` fine. `./examples` and `./blueprints` as well as test should always be on latest hashes

ijc commented:
I fixed `tests`. `examples` and `blueprints`were ok already.

Runes of this form were helpful:
`git grep -H -A2 linuxkit/init: -- examples | sed -e 's/.*\.yml[:-]  - \(.*\)/\1/g' | sort | uniq -c`

Part of the problem is that if someone forgets to bump one of these somehow then future bumps using `scripts/update-component-sha.sh` won't manage to update them. Perhaps we need a better script to sync `linuxkit.yml` to the places we think it should be sync'd

rn commented:
Define CONTAINERD_REPO alongside CONTAINERD_COMMIT in tools/alpine seems to have been update after the update to `containerd`. Wouldn't that mean we also need to change the alpine base hash used by them?

ijc commented:
@rn the update was deferred to the next commit when the referenced tree/hash was actually changed (I mentioned this in the commit log, but of course GH UI doesn't encourage to look at that).

riyazdf commented:
Post facto LGTM - thanks @ijc!

> On Jul 13, 2017, at 4:58 AM, Rolf Neugebauer <notifications@github.com> wrote:
> 
> @rn approved this pull request.
> 
> ah, missed that
> 
> ―
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
